go/common/tribool/tribool.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package tribool
+
+// Tribool is a three-valued logic type.
+// It can be either True, False or Unknown.
+// It is useful for representing the result of a comparison
+// when the comparison is not possible or could not be determined.
+type Tribool struct {
+	value byte
+}
+
+// Unknown returns true if the value is unknown.
+func (t Tribool) Unknown() bool {
+	return t.value == 0
+}
+
+// False returns true if the value is false.
+func (t Tribool) False() bool {
+	return t.value == 1
+}
+
+// True returns true if the value is true.
+func (t Tribool) True() bool {
+	return t.value == 2
+}
+
+// NewUnknown creates a new Tribool with unknown value.
+func NewUnknown() Tribool {
```

Comment:
ok, make sense. 

Notice that I did not use "Is" prefix for getters. It seems common in goalng to call a setter  `True()` not `IsTrue() `etc. 

I used `New` for constructors as it seems to be also common, but in this case I thin you are right to avoid this prefix